pyanaconda/payload/manager.py

```diff
@@ -0,0 +1,238 @@
+# Class for management payload threading.
+#
+# Copyright (C) 2019  Red Hat, Inc.
+#
+# This copyrighted material is made available to anyone wishing to use,
+# modify, copy, or redistribute it subject to the terms and conditions of
+# the GNU General Public License v.2, or (at your option) any later version.
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY expressed or implied, including the implied warranties of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General
+# Public License for more details.  You should have received a copy of the
+# GNU General Public License along with this program; if not, write to the
+# Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+# 02110-1301, USA.  Any Red Hat trademarks that are incorporated in the
+# source code or documentation are not subject to the GNU General Public
+# License and may only be used or replicated with the express permission of
+# Red Hat, Inc.
+#
+
+import threading
+
+from pyanaconda.core.constants import THREAD_STORAGE, THREAD_PAYLOAD, THREAD_PAYLOAD_RESTART, \
+    THREAD_WAIT_FOR_CONNECTING_NM
+from pyanaconda.core.i18n import _, N_
+from pyanaconda.threading import threadMgr, AnacondaThread
+from pyanaconda.payload import PackagePayload
+from pyanaconda.payload.errors import PayloadError
+
+from pyanaconda.anaconda_loggers import get_module_logger
+
+log = get_module_logger(__name__)
+
+
+__all__ = ["payloadMgr"]
+
+
+class PayloadManager(object):
```

Comment:
I think that the payload manager could be already replaced with a task, because it provides many of the required features by default. The same task can be later used in the DBus Payload module.

Reply:
Yeah could be done in a separate PR.